src/qibo/backends/qulacs.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import numpy as np
+
+from qibo import __version__
+from qibo.backends import NumpyBackend
+from qibo.result import CircuitResult, QuantumState
+
+
+class QulacsBackend(NumpyBackend):
+
+    def __init__(self):
+        super().__init__()
+        import qulacs  # pylint: disable=import-error
+        from qulacs import (  # pylint: disable=no-name-in-module, import-error
+            QuantumCircuitSimulator,
+            converter,
+        )
+
+        self.qulacs = qulacs
+        self.simulator = QuantumCircuitSimulator
+        self.converter = converter
+        self.name = "qulacs"
+        self.versions = {"qibo": __version__, "qulacs": qulacs.__version__}
+        self.device = "CPU"
+
+    def circuit_to_qulacs(
+        self, circuit: "qibo.Circuit"
+    ) -> "qulacs.QuantumCircuit":  # pylint: disable=no-member
+        """
+        Converts a qibo circuit in a qulacs circuit.
+
+        Args:
+            circuit (:class:`qibo.models.circuit.Circuit`): Input circuit to convert.
+
+        Returns:
+            qulacs.QuantumCircuit: The converted qulacs circuit.
+        """
+        qasm_str = re.sub("^//.+\n", "", circuit.to_qasm())
+        qasm_str = re.sub(r"creg\s.+;", "", qasm_str)
+        qasm_str = re.sub(r"measure\s.+;", "", qasm_str)
+        circ = self.converter.convert_QASM_to_qulacs_circuit(qasm_str.splitlines())
+        return circ
```

Comment:
If `converter` were not an attribute of `self` (but just an imported variable), this could be a top-level function, no need of a method :)